pkg/ingress/compare.go

```diff
@@ -42,7 +43,7 @@ func (c *Controller) CompareResources(ctx context.Context) error {
 		consumerMapA6    = make(map[string]string)
 	)
 	// watchingNamespace == nil means to monitor all namespaces
-	if c.watchingNamespace == nil {
+	if !validation.HasValueInSyncMap(c.watchingNamespace) {
```

Comment:
This check is not safe. The result is only correct at a very specific point-in-time, which means the map might contains some elements after you check it. Please assess the impacts. If necessary, use a mutex.

Reply:
I think it is not necessary, because the `sync.Map` in `HasValueInSyncMap ` is concurrency safe, which use `atomic.Value` and `Mutex` inside.

Reply:
@tokers I hope we can reach an agreement in this regard.

Reply:
OK, I check out the code base I found there is only place use the `CompareResources` and at that time the controllers are not launched.

So let's just add some comments on this method so that once this method is called in the runtime, writers won't make a mistake.